projects/packages/woocommerce-analytics/src/class-woo-analytics-trait.php

```diff
@@ -397,14 +397,17 @@ public function maybe_start_session() {
 
 			$session_expiration = $this->get_session_expiration_time();
 			$event_js           = $this->process_event_properties( 'woocommerceanalytics_session_started' );
-			$cookie_js          = "
-            const sessionData = JSON.stringify({
-                    session_id: '$this->session_id',
-                    landing_page: encodeURIComponent('$this->landing_page'),
```

Comment:
Previously, if somehow `loading page` is not sanitized, then the malicious URL gets interpolated directly into the JavaScript:
  
```js
  const sessionData = JSON.stringify({
      session_id: 'some-uuid',
      landing_page:
  encodeURIComponent('https://yoursite.com/shop/?test=');alert('XSS');//'),
      expires: '...',
  });
```

Now, we use `wp_json_encode` to avoid any potential attacks.